var/spack/repos/builtin/packages/cmake/package.py

```diff
@@ -63,7 +63,8 @@ class Cmake(Package):
     depends_on('qt',             when='+qt')
     depends_on('python@2.7.11:', when='+doc', type='build')
     depends_on('py-sphinx',      when='+doc', type='build')
-    depends_on('openssl',        when='+openssl')
+    depends_on("openssl@:1.0.99", when='@:3.6.9+openssl')
+    depends_on("openssl", when='@3.7:+openssl')
```

Comment:
I have a feeling I've been down this road before ([and you were there, and you, and you](https://www.youtube.com/watch?v=BZSb0JCWcXk&feature=youtu.be&t=1m4s)).

I'll take this version for a test drive...

@davydden -- any objections?